Sources/Filters/General/ClipClosedSurface/index.js

```diff
@@ -0,0 +1,1156 @@
+import macro from 'vtk.js/Sources/macros';
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+import vtkCellArray from 'vtk.js/Sources/Common/Core/CellArray';
+import vtkDataArray from 'vtk.js/Sources/Common/Core/DataArray';
+import { VtkDataTypes } from 'vtk.js/Sources/Common/Core/DataArray/Constants';
+import vtkPoints from 'vtk.js/Sources/Common/Core/Points';
+import vtkDataSetAttributes from 'vtk.js/Sources/Common/DataModel/DataSetAttributes';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData';
+import vtkContourTriangulator from 'vtk.js/Sources/Filters/General/ContourTriangulator';
+
+import Constants from './Constants';
+import CCSEdgeLocator from './ccsEdgeLocator';
+
+const { vtkErrorMacro, capitalize } = macro;
+const { ScalarMode } = Constants;
+
+function vtkClipClosedSurface(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkClipClosedSurface');
+
+  publicAPI.getMTime = () =>
+    model.clippingPlanes.reduce(
+      (a, b) => (b.getMTime() > a ? b.getMTime() : a),
+      model.mtime
+    );
+
+  /**
+   * Take three colors as doubles, and convert to unsigned char.
+   *
+   * @param {Number} color1
+   * @param {Number} color2
+   * @param {Number} color3
+   * @param {Number[3][3]} colors
+   */
+  function createColorValues(color1, color2, color3, colors) {
+    const dcolors = [color1, color2, color3];
+    const clamp = (n, min, max) => Math.min(Math.max(n, min), max);
+
+    for (let i = 0; i < 3; i++) {
+      for (let j = 0; j < 3; j++) {
+        colors[i][j] = Math.round(clamp(dcolors[i][j], 0, 1) * 255);
+      }
+    }
+  }
+
+  /**
+   * Point interpolation for clipping and contouring, given the scalar
+   * values (v0, v1) for the two endpoints (p0, p1).  The use of this
+   * function guarantees perfect consistency in the results.
+   *
+   * @param {vtkPoints} points
+   * @param {vtkDataArray} pointData
+   * @param {CCSEdgeLocator} locator
+   * @param {Number} tol
+   * @param {Number} i0
+   * @param {Number} i1
+   * @param {Number} v0
+   * @param {Number} v1
+   * @param {Number} i
+   * @returns {Number}
+   */
+  function interpolateEdge(points, pointData, locator, tol, i0, i1, v0, v1, i) {
+    // This swap guarantees that exactly the same point is computed
+    // for both line directions, as long as the endpoints are the same.
+    if (v1 > 0) {
+      // eslint-disable-next-line no-param-reassign
+      [i0, i1] = [i1, i0];
+      // eslint-disable-next-line no-param-reassign
+      [v0, v1] = [v1, v0];
+    }
+    // After the above swap, i0 will be kept, and i1 will be clipped
+
+    // Check to see if this point has already been computed
+    let node;
+    // eslint-disable-next-line no-param-reassign
```

Comment:
it's not great to disable this eslint rule. How about:
`let [node, edgeId] = ...`

Reply:
Problem here is `node` is a `const` and `edgeId` needs to be `let`

Reply:
Nvm. I got it now!